fvm/tests/default_kernel/dummy.rs

```diff
@@ -0,0 +1,267 @@
+use std::mem::ManuallyDrop;
+use std::sync::{Arc, Weak};
+
+use anyhow::Context;
+use derive_more::{Deref, DerefMut};
+use fvm::call_manager::{Backtrace, CallManager, FinishRet, InvocationResult};
+use fvm::gas::{Gas, GasCharge, GasTracker};
+use fvm::kernel;
+use fvm::machine::{Engine, EngineConfig, MachineContext, NetworkConfig};
+use fvm::state_tree::{ActorState, StateTree};
+use fvm_ipld_blockstore::{Blockstore, MemoryBlockstore};
+use fvm_ipld_encoding::CborStore;
+use fvm_shared::actor::builtin::Manifest;
+use fvm_shared::address::Address;
+use fvm_shared::state::StateTreeVersion;
+// use fvm_shared::version::NetworkVersion;
+use multihash::Code;
+
+use super::*;
+use crate::DummyExterns;
+
+/// this is essentially identical to DefaultMachine, but has pub fields for reaching inside
```

Comment:
If it's nearly identical, this could be a bit annoying to maintain. Any reason not to put these unit tests inside the default kernel module? That'll give them access to private things.

Reply:
Hm. Although, it doesn't look all that identical.

Reply:
it is outside of not using a buffered blockstore, non-parameterized types, non complete transfer fn and create actor fn

Reply:
I would argue that `DummyX` structs should stay independent as its supposed to be a testing backboard that (hopefully) unexpected changes in *just the kernel* get caught by tests which means having minimal dependencies on other parts of code. Expected changes should need refactoring, but keeping that as explicit should be correct.

I could move unit tests inside of the default kernel module and make fields `pub(crate)` so I could access values directly, though it would be a fairly hefty file after adding of all the tests for it.

Reply:
i could do 
```
kernel
- tests 
  - mod.rs
  - ops.rs
  - x.rs
- mod.rs
- default.rs
```
 
```rust
// in mod.rs
#[cfg(test)]
mod tests;
``` 
```rust
// in default.rs
struct DefaultKernel {
  pub(crate): ...
}
```
which would let me avoid all the dealing with external call manager shenanigans that i'm doing right now, though theoretically with the current setup it would be able to test different kernel implementations with few(er) modifications.

Reply:
Eh, we're probably going to need to instrument the test machine quite a bit... so we might as well keep it separate. We're not testing it.